### PR TITLE
[Simd] Implement extra convert, trunc, demote and promote ops for interpreter

### PR DESCRIPTION
**Pull request by MaxGraey**
```ts
f64x2.convert_low_i32x4_s
f64x2.convert_low_i32x4_u
i32x4.trunc_sat_f64x2_s_zero
i32x4.trunc_sat_f64x2_u_zero
f32x4.demote_f64x2_zero
f64x2.promote_low_f32x4
```

### CODE REVIEW

**Comment by MaxGraey** [comment]
@tlively Could you take a look please?

File: src/wasm/literal.cpp [hunk 1]
```diff
@@ -2388,6 +2388,16 @@ Literal extend(const Literal& vec) {
   return Literal(result);
 }
 
+template<LaneOrder Side> Literal extendF32(const Literal& vec) {
+  LaneArray<4> lanes = vec.getLanesF32x4();
+  LaneArray<2> result;
+  for (size_t i = 0; i < 2; ++i) {
+    size_t idx = (Side == LaneOrder::Low) ? i : i + 2;
+    result[i] = Literal((double)(float)lanes[idx].getf32());
```
Comment (tlively) [inline src/wasm/literal.cpp:2396]:
```suggestion
    result[i] = Literal((double)lanes[idx].getf32());
```

File: src/wasm/literal.cpp [hunk 2]
```diff
@@ -2388,6 +2388,16 @@ Literal extend(const Literal& vec) {
   return Literal(result);
 }
 
+template<LaneOrder Side> Literal extendF32(const Literal& vec) {
```
Comment (tlively) [inline src/wasm/literal.cpp:2391]: An alternative to adding a new template here would be to make the `geti32` a template parameter of `extend`, but this seems fine too.

Comment (MaxGraey) [reply src/wasm/literal.cpp:2391]: I initially thought of unifying the "extend" too, but SIMD is already standardized and this special case is only needed once

**Review by tlively — APPROVED** [review]

**Pull request merged**